iceoryx_posh/include/iceoryx_posh/internal/popo/ports/base_port.hpp

```diff
@@ -81,7 +80,7 @@ class BasePort
     MemberType_t* getMembers() noexcept;
 
   private:
-    MemberType_t* m_basePortDataPtr{nullptr};
+    MemberType_t* m_basePortDataPtr;
```

Comment:
Why did you remove the value initialisation?

Reply:
As this is provided in the c'tor, it is set in the member initializer list. So it is somehow redundant and as far as I remember I realized it as QAC came up with a warning. Here it also gives you a feeling that you can end up with an object that has nullptr for m_basePortDataPtr. Which is not the case because it comes with cxx::not_null in the c'tor

Reply:
didn't we decide on storing references?

Reply:
The reference refactoring will be done in another PR. This affects also some old modules and I didn't want to blow up this PR

Reply:
> didn't we decide on storing references?

@elBoberido this is not yet possible. If you have references as members in a class the class is neither move- nor copy-able and at the moment we store all port data in a vector. We have to refine this first and need to use the upcoming `cxx::list` before we can go in this direction.

Reply:
@elfenpiff Sure, we have to collect the alternatives and agree on the best solution. I already had some discussions with @elBoberido

Reply:
only move assignment is not possible, move c'tors still work